commands/server/start.go

```diff
@@ -72,17 +72,28 @@ func StartCmd(gen AppGenerator, logger log.Logger, home string, args []string) e
 	if err != nil {
 		return errors.Wrap(err, "failed to create a listener")
 	}
+
 	svr.SetLogger(logger.With("module", "abci-server"))
-	err = svr.Start()
-	if err != nil {
-		return errors.Wrap(err, "failed to start a server")
-	}
+
 	done := make(chan bool)
-	cmn.TrapSignal(logger, func() {
+	cleanupCallback := func() {
 		// Cleanup
 		_ = svr.Stop()
 		done <- true
-	})
+	}
+
+	cmn.TrapSignal(logger, cleanupCallback)
```

Comment:
[`TrapSignal`](https://github.com/tendermint/tendermint/blob/e9c9c558d7a347b37c9ac6ca770107f84ad24696/libs/common/os.go#L17) (what a terrible name) is handling only SIGINT and SIGTERM. I understand we always want to execute `cleanupCallback`.
Why no instead of trap signal and defer+recover, always cleanup with `defer cleanupCallback()`?

Reply:
This is tendermint standard stuff.... (`TrapSignal` and naming) `kill` would terminate the process with a signal that would never call the defer function. (it never returns, just dies)

I think this is good enough, as it works for tendermint, and unless we want to put a lot of thought into it... "good enough for mainnet"